app/api/odm/ModelWithPermissions.ts

```diff
@@ -150,7 +150,11 @@ export class ModelWithPermissions<T> extends OdmModel<T> {
   async getById(id: any, select?: string) {
     const user = permissionsContext.getUserInContext();
     const doc = await this.db.findOne(
-      appendPermissionQuery({ _id: id || null }, AccessLevels.READ, user),
+      appendPermissionQuery(
+        { _id: id || null } as PermissionsUwaziFilterQuery<T>,
```

Comment:
@RafaPolit im pretty sure this is not needed, we should be able to pass the proper object. what is the reason ?, i think we typed wrongly the DataModelType?

Reply:
It is not needed with Typescirpt 4.3, it does not give any error.  It fails on 3.9.